Source/WebKit/UIProcess/mac/TextCheckerMac.mm

```diff
@@ -395,6 +399,23 @@ static bool shouldAutomaticDashSubstitutionBeEnabled()
             result.type = TextCheckingType::Correction;
             result.range = resultRange;
             result.replacement = [incomingResult replacementString];
+            if ([incomingResult respondsToSelector:@selector(detail)]) {
+                NSDictionary *incomingDetail = [incomingResult detail];
+                if (incomingDetail) {
+                    result.details.reserveInitialCapacity(1);
+                    GrammarDetail detail;
+                    NSValue *detailRangeAsNSValue = [incomingDetail objectForKey:NSGrammarRange];
+                    ASSERT(detailRangeAsNSValue);
```

Comment:
Weird that these two blocks of code
A) have wildly different paragraphing
B) look very similar otherwise

Can we factor them out? I didn't look carefully.

Also not entirely sure reserveInitialCapacity/uncheckedAppend is worth it for 1 element, but maybe! (Also, not sure it's right, since your `result.details` seems to exist before you `reserveInitialCapacity`, are you sure it's never been used before?)

Reply:
Oh, I see, the thing was constructed right there. So, still don't think it's necessary, but it is fine.